api/reporting/queries/live/restructured_results/latest_users_structured.sql

```diff
@@ -0,0 +1,40 @@
+SELECT
+    TIMESTAMP_MILLIS(u.snapshot_timestamp) AS snapshot,
```

Comment:
Nifty!

I worry a small bit about "give a mouse a cookie" here, where there may become an expectation that *all* of our underlying data is structured this way, when it's really a manually crafted facade. Not necessarily a bad thing – it's just worth being clear about what this is and isn't.

We also got a pretty signal early on (from the PDR folks I believe) that their Tableau connector wouldn't be able to make use of BigQuery's nested / struct rows. Not sure whether that's still the case.

Anyhow, those are just points to keep in mind. Seems useful and worth keeping around, at least as a POC of how one might add some structure to the BQ version of this reporting data.

Reply:
Yeah, this is  more for Shimon and other internal customers. 

Look at  it like UI; we have a workspaces page, but people don't expect to be able  to view all of our objects in that format.